baseapp/abci.go

```diff
@@ -45,6 +38,13 @@ func (app *BaseApp) InitChain(req abci.RequestInitChain) (res abci.ResponseInitC
 
 	res = app.initChainer(app.deliverState.ctx, req)
 
+	// Store the consensus params in the BaseApp's paramstore. Note, this must be
+	// done after the deliver state and context have been set as it's persisted
+	// to state.
+	if req.ConsensusParams != nil {
+		app.StoreConsensusParams(app.deliverState.ctx, req.ConsensusParams)
+	}
+
```

Comment:
Was this intentionally moved after `app.initChainer` ? Wondering if this has something to do with the panics you're finding...

Also, what are you doing to reproduce the panic behavior?

Reply:
StoreConsensusParams calls into the KV stores, but they need to be loaded beforehand. The stores (in this iteration of the PR) are loaded inside `app.initChainer`, hence putting these lines after `app.initChainer`.

To test, I'm running the test I wrote TestSimappInitialHeight.